server/controllers/profile.js

```diff
@@ -0,0 +1,91 @@
+const ObjectId = require('mongoose').Types.ObjectId;
+const asyncHandler = require("express-async-handler");
+const Profile = require("../models/Profile");
+const User = require("../models/User");
+
+// @route POST /profile
+// @Given parameters passed in, create a profile.
+exports.createProfile = asyncHandler(async (req, res, next) => {
+  let profileData = req.body;
+  let profile = new Profile(profileData);
+
+  try {
+    const newProfile = await profile.save();
+    res.status(200).json({
+      success: {
+        profile: newProfile,
+      },
+    });
+  } catch (e) {
+    res.status(500);
+    throw new Error(e.message);
+  }
+});
+
+// @route PUT /profile/:id
+// @Given an ID and new parameters, update the profile
+exports.updatedProfile = asyncHandler(async (req, res, next) => {
+  const id = req.params.id;
+  let profile = req.body;
```

Comment:
this can be `const`. Also is it the entire body? not something like `req.body.profile`?

Reply:
changed to req.body.profile